great_expectations/experimental/datasources/interfaces.py

```diff
@@ -407,6 +415,7 @@ def add_asset(self, asset: _DataAssetT) -> _DataAssetT:
         # The setter for datasource is non-functional, so we access _datasource directly.
         # See the comment in DataAsset for more information.
         asset._datasource = self
+        asset._build_data_connector()
```

Comment:
This isn't appropriate, not all assets have a data connector. This is another reason to keep initialization in the constructor.

For ZEP, data connectors are an internal implementation detail of certain GX assets. We need to limit the contract surface and keep it that way